### PR TITLE
[#5175] handle all exceptions in file list endpoint

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: hs_core/views/resource_rest_api.py [hunk 1]
```diff
@@ -768,8 +768,10 @@ def list(self, request, *args, **kwargs):
                 try:
                     resource_file_info_list.append(self.resourceFileToListItem(f))
                 except SessionException as err:
-                    # primarily this exception will be raised if the file is not found in iRODS
                     logger.error(f"Error for file {f.storage_path} from iRODS: {err.stderr}")
+                except Exception as err:
+                    # primarily this exception will be raised if the file is not found in iRODS
+                    logger.error(f"Error for file {f.storage_path} from iRODS: {str(err)}")
```
Comment (devincowan) [inline hs_core/views/resource_rest_api.py:774]: maybe just combine both `except` blocks into a single one? Unless I'm forgetting that `SessionException.stderr` is significantly different than simply `str(err)`

**Review by devincowan — DISMISSED** [review]

File: hs_core/views/resource_rest_api.py (hunk 1 above)
Comment (pkdash) [inline hs_core/views/resource_rest_api.py:774]: The use of `f.storage_path `results in 1 db query. If we want we should be able to avoid it.

Comment (devincowan) [reply hs_core/views/resource_rest_api.py:774]: Got it, that makes sense.
Why not target `ValidationError` specifically instead of general `Exception` ?

Comment (pkdash) [reply hs_core/views/resource_rest_api.py:774]: I think it is probably different.

Comment (devincowan) [reply hs_core/views/resource_rest_api.py:774]: sounds good

Comment (pkdash) [reply hs_core/views/resource_rest_api.py:774]: Fixed it.

**Review by devincowan — APPROVED** [review]

**Comment by devincowan** [comment]
Closing as we will use a rebase of this branch (https://github.com/hydroshare/hydroshare/pull/5179) to release as hotfix 2.8.1

**Pull request closed without merging**